UniversityApp.BussinessLogic/Services/StudentsService.cs

```diff
@@ -99,34 +99,35 @@ public async Task<List<GradeDomainModel>> GetStudentGrades(int id)
             return result;
         }
 
-        public async Task<int?> AddStudentGrade(int id, GradeDomainModel grade)
+        public async Task<int?> AddStudentGrade(GradeDomainModel grade)
         {
-            var student = await _studentsRepository.GetStudent(id);
-
-            if(student == null)
+            if (!(await DoesStudentExists(grade.StudentId)))
```

Comment:
I think it would be more readable when:
`
var doesStudentExist = await DoesStudentExists(grade.StudentId);

if (!doesStudentExist){
     [....]
}
`

Reply:
done :)